### PR TITLE
refactor: bytes signature library

### PR DESCRIPTION
**Pull request by fedealconada**
Refactors `BytesSignature.sol`. WDYT @rrecuero ?

### CODE REVIEW

File: test/BytesSignature.t.sol [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        bytes memory signature1 =
+            hex"ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff01";
+        bytes memory signature2 =
+            hex"aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaabbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb1c";
+        bytes memory fullSignature = abi.encodePacked(signature1, signature2);
+
+        bytes[] memory extracted = ByteSignature.extractSignatures(fullSignature, 2);
+        assertEq(extracted[0], signature1, "The first signature does not match the original signature.");
+        assertEq(extracted[1], signature2, "The second signature does not match the original signature.");
+    }
```
Comment (rrecuero) [inline test/BytesSignature.t.sol:27]: can we add another test with 3 just in case? It should work

**Comment by rrecuero** [comment]
It is much better generalized this way. I started with one or two and added three and never changed it. 🙏

**Review by rrecuero — CHANGES_REQUESTED** [review]
A small thing to add one more test with 3 sigs

File: test/BytesSignature.t.sol (hunk 1 above)
Comment (fedealconada) [reply test/BytesSignature.t.sol:27]: i've added a fuzz test so we test any amount of signatures

**Pull request merged**